lib/llvmopencl/LLVMUtils.cc

```diff
@@ -726,10 +726,16 @@ llvm::Type *SizeT(llvm::Module *M) {
   return IntegerType::get(M->getContext(), AddressBits);
 }
 
-bool isWorkitemFunctionWithOnlyCompilerExpandableCalls(const llvm::Function &F) {
-  Instruction::const_use_iterator UI = F.use_begin(), UE = F.use_end();
-  for (; UI != UE; ++UI) {
-    llvm::CallInst *Call = dyn_cast<llvm::CallInst>(UI->getUser());
+bool isWorkitemFunctionWithOnlyCompilerExpandableCalls(
+    const llvm::Function &F) {
+
+  if (F.getName() != GID_BUILTIN_NAME && F.getName() != GS_BUILTIN_NAME &&
+      F.getName() != GROUP_ID_BUILTIN_NAME && F.getName() != LID_BUILTIN_NAME &&
+      F.getName() != LS_BUILTIN_NAME && F.getName() != NGROUPS_BUILTIN_NAME)
+    return false;
+
+  for (const auto &U : F.uses()) {
+    llvm::CallInst *Call = dyn_cast<llvm::CallInst>(U.getUser());
```

Comment:
My two cents. The type is spelled in the dyn_cast:
```suggestion
    auto *Call = dyn_cast<llvm::CallInst>(U.getUser());
```

Reply:
Yep, I think using auto should not be automatic (for readability), but mainly help with more complex types. But more of a matter of taste, I guess.